components/brave_webtorrent/browser/net/brave_torrent_redirect_network_delegate_helper.cc

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_webtorrent/browser/net/brave_torrent_redirect_network_delegate_helper.h"
+
+#include "base/strings/strcat.h"
+#include "base/strings/string_util.h"
+#include "brave/common/extensions/extension_constants.h"
+#include "extensions/common/constants.h"
+#include "net/http/http_content_disposition.h"
+#include "net/http/http_response_headers.h"
+#include "net/url_request/url_request.h"
+
+namespace {
+
+bool FileNameMatched(const net::HttpResponseHeaders* headers) {
+  std::string disposition;
+  if (!headers->GetNormalizedHeader("content-disposition", &disposition)) {
+    return false;
+  }
+
+  net::HttpContentDisposition cd_headers(disposition, std::string());
+  if (base::EndsWith(cd_headers.filename(), ".torrent",
+        base::CompareCase::INSENSITIVE_ASCII) ||
+      base::EndsWith(cd_headers.filename(), ".torrent\"",
+        base::CompareCase::INSENSITIVE_ASCII)) {
+    return true;
+  }
+
+  return false;
+}
+
+bool URLMatched(net::URLRequest* request) {
+  return base::EndsWith(request->url().spec(), ".torrent",
+      base::CompareCase::INSENSITIVE_ASCII);
+}
+
+bool IsTorrentFile(net::URLRequest* request,
+    const net::HttpResponseHeaders* headers) {
+  std::string mimeType;
+  if (!headers->GetMimeType(&mimeType)) {
+    return false;
+  }
+
+  if (mimeType == "application/x-bittorrent") {
```

Comment:
nit: pls add this to `common/network_constants.h` and use the constant here and in tests